### DISCUSSION TITLE
Import data from Hammond

### DISCUSSION THREAD

**Discussion by ThisIsQasim** [hargata/lubelog#578] [Ideas]
I have been using Hammond for a few months. It works alright but lubelogger seems more feature complete in comparison so I would like to migrate. Would it be possible to have any easy way to migrate the data? I think my only option right now is the csv import.

**Comment by Pineapplesandwich** [comment]
I would also like to have this ability but it does not appear that Hammond supports exporting easily..

**Comment by ThisIsQasim** [comment]
I think a migration script should be possible or probably the database file from hammond can be used to at least migrate the records.

**Reply by hargata** [reply]
I've looked at [Hammond's code](https://github.com/akhilrex/hammond/blob/master/server/db/db.go), based on what I see this seems like the most plausible method. You'd have to write a script that utilizes SQLite to open up `hammond.db`, read the tables and then utilize LubeLogger's API to post those records over, alternatively you can also write the records to a CSV file that can be imported into LubeLogger.

The only issue I can see is that LubeLogger's API doesn't have an endpoint to create vehicles or users, just records for a specific vehicle, so you'd have to re-create the vehicles or users by hand first.

**Reply by ThisIsQasim** [reply]
I don’t mind creating the vehicles and the users. It’s the car records that need to be imported.

**Comment by ThisIsQasim** [comment]
In case anyone wanders here. I ended up manually creating the vehicles in Lubelogger and creating a csv for the fuel records from the data in Hammond and importing that. I had to exec into the Hammond container, install the sqlite client, get the vehicle UUID from the `user_vehicles` table and then create a csv by querying the fuel records.

Please note that the `strftime` date format depends on the `LC_ALL` set in Lubelogger.
```
docker-compose exec hammond -- sh
apk add sqlite
sqlite3 /config/hammond.db

.mode csv

select * from user_vehicles;

select strftime('%m/%d/%Y', date),
	odo_reading,
	printf('%.3f', fuel_quantity),
	printf('%.2f', total_amount),
	CASE WHEN is_tank_full = 1 THEN 'True' ELSE 'False' END,
	CASE WHEN has_missed_fillup = 1 THEN 'True' ELSE 'False' END,
	comments
	 from fillups where vehicle_id='1b27bb2d-3cc3-4e78-bb1a-5b0748c4f774';
```

**Reply by geomule** [reply]
I ran this command `docker-compose exec hammond -- sh` from within the folder where I have the Hammond yaml file and got this error `OCI runtime exec failed: exec failed: unable to start container process: exec: "--": executable file not found in $PATH`. Note that the `docker compose -f docker-compose.yml up -d --remove-orphans` works normally from that folder.

EDIT: This does work `docker exec -it hammond sh`. Although now trying to work out where the various SELECT commands has dumped the data. As `.mode csv` is activated, I woudl assume that the data is sitting in all the CSV files. I was expecting to see them in the directory of the yaml file from where I ran the shell but nada.

**Reply by ThisIsQasim** [reply]
My bad. I had it deployed on Kubernetes so copied the command from there. Docker does not expect the `--`
